react-validated-input-component/src/validated-input.jsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+
+class FormValidator extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      password: '',
+      message: 'A Password is required',
+      validator: 'times'
+    };
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleChange(event) {
+    const passLength = this.state.password.length;
+    this.setState({
+      password: event.target.value
+    });
+    if (passLength >= 7) {
+      this.setState({
+        message: '',
+        validator: 'check'
+      });
+    } else if (passLength <= 7) {
+      this.setState({
+        message: 'Password is too short',
+        validator: 'times'
+      });
+    }
```

Comment:
Instead of checking the state (which is one step behind), check the `event.target.value`